oauthproxy.go

```diff
@@ -1049,3 +1072,23 @@ func (p *OAuthProxy) ErrorJSON(rw http.ResponseWriter, code int) {
 	rw.Header().Set("Content-Type", applicationJSON)
 	rw.WriteHeader(code)
 }
+
+func (p *OAuthProxy) validateGroups(groups []string) bool {
+	if len(p.AllowedGroups) == 0 {
+		return true
+	}
+
+	m := map[string]bool{}
+
+	for _, g := range p.AllowedGroups {
+		m[g] = true
+	}
+
+	for _, g := range groups {
+		if m[g] {
+			return true
+		}
+	}
```

Comment:
Not a huge fan of single letter variables in this case, what does m stand for? 

I know this is a nit but for future maintainability, can we use real variable names?

Reply:
Agreed, I changed it, TBH single letter variables bother me too but looking around the place I spotted a bunch and have seen a preference in other Go projects for them so just went with what I saw, this is changed!

Reply:
Agreed, I changed it, TBH single letter variables bother me too but looking around the place I spotted a bunch and have seen a preference in other Go projects for them so just went with what I saw, this is changed!

> Just given another pass, added a few more comments, mainly nits, but one is an actual concern, I'd like to make sure the group extraction process is thoroughly tested as I think at the moment it could unintentionally extract groups if the groups claim is set to something other than `groups`

Thanks just fixed those things up, the mentioned `groups` issue, I had simply missed removing that in a previous commit, it should now not be unmarshaled anymore.